ci_framework/roles/ci_network/tasks/apply.yml

```diff
@@ -0,0 +1,43 @@
+---
+- name: Assert network configuration is valid
+  block:
+    - name: VLAN data are present
+      when:
+        - network.key is match('^vlan.*')
+      ansible.builtin.assert:
+        that:
+          - network.value.parent_iface is defined
+        msg: "parent_iface is needed for VLAN interface"
+
+    - name: IPv4 is defined
+      ansible.builtin.assert:
+        that:
+          - network.value.ip4 is defined
+        msg: "ip4 is needed for any interface"
+
+- name: "Apply network {{ network.key }}"
+  become: true
+  when:
+    - network.value.config_nm | default(true) | bool
+  community.general.nmcli:
+    autoconnect: true
+    state: "{{ network.value.state | default('present') }}"
+    conn_name: "{{ network.value.connection | default(network.key) }}"
+    ifname: "{{ network.value.iface }}"
```

Comment:
Not needed for VLANs, not sure how this module behaves if passed when type is vlan.

Reply:
hmm apparently it's working fine, since molecule is green

Reply:
```
The ifname argument is mandatory for all connection types except bond, team, bridge, vlan and vpn.

This parameter defaults to conn_name when left unset for all connection types except vpn that removes it.
```
Guess we can therefore safely keep it as-is.